hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/KeyValueContainerUtil.java

```diff
@@ -365,6 +367,20 @@ private static void populateContainerMetadata(
     ContainerInspectorUtil.process(kvContainerData, store);
   }
 
+  private static void populateContainerFinalizeBlock(
+      KeyValueContainerData kvContainerData, DatanodeStore store)
+      throws IOException {
+    Table<String, FinalizeBlockList> finalizeBlocksTable =
+        store.getFinalizeBlocksTable();
+
+    FinalizeBlockList finalizeBlockList =
+        finalizeBlocksTable.get(kvContainerData.getFinalizeBlockKey());
```

Comment:
Here it looks a little strange. Assuming you fetch FinalizeBlockList regardless of their containers, you would get all finalize blocks, but FinalizeBlockList is a list of Long, it is unable to tell what containers the blocks belong too.

Reply:
I suspect you want to use KeyValueContainerData.getBlockKey() to add prefix for different containers.

Reply:
It would have loaded only container specific list as based on previous comment format key returns containerId as key in schema v3. Now i have changed key structure as previous comment..